### PR TITLE
 improve code's style with some doc

### PR DESCRIPTION
**Pull request by lanzhiheng**
# Summary

Improve code's style for `hash_with_indifferent_access.rb` and write some doc for it.

### CODE REVIEW

**Comment by lanzhiheng** [comment]
@eileencodes  The building flow was broken, I have no idea how to fix it, can you give me a hand? 😿  

image

**Comment by tgxworld** [comment]
@lanzhiheng The changes here are generally cosmetic in nature and most likely wouldn't be accepted. 

> [Changes that are cosmetic in nature and do not add anything substantial to the stability, functionality, or testability of Rails will generally not be accepted](https://guides.rubyonrails.org/contributing_to_ruby_on_rails.html#write-your-code)

https://guides.rubyonrails.org/contributing_to_ruby_on_rails.html#write-your-code

File: activesupport/lib/active_support/hash_with_indifferent_access.rb [hunk 1]
```diff
@@ -60,13 +60,16 @@ def with_indifferent_access
...
+    # who was defined in `activesupport/lib/active_support/core_ext/hash/indifferent_access.rb`.
+    # If `self.is_a?(HashWithIndifferentAccess)` is true, we don't need to instantiate again.
     def nested_under_indifferent_access
       self
     end
 
     def initialize(constructor = {})
       if constructor.respond_to?(:to_hash)
-        super()
+        super
```
Comment (jonathanhefner) [inline activesupport/lib/active_support/hash_with_indifferent_access.rb:72]: This is actually subtly different.  `super()` calls the superclass's `initialize` without any args, but `super` calls the superclass's `initialize` with all defined args to, i.e. `super(constructor)` in this case.

**Comment by jonathanhefner** [comment]
Yes, @tgxworld is correct.  Additionally, I think this is a good example of how cosmetic changes can be tricky.  See [my comment](https://github.com/rails/rails/pull/39945/files#r462371875).

File: activesupport/lib/active_support/hash_with_indifferent_access.rb (hunk 1 above)
Comment (lanzhiheng) [reply activesupport/lib/active_support/hash_with_indifferent_access.rb:72]: Oh, that is my fault. Thanks for your guide.

**Pull request closed without merging**